web/packages/core/src/ruffle-player.ts

```diff
@@ -1433,6 +1482,25 @@ export class RufflePlayer extends HTMLElement {
     setIsExtension(isExtension: boolean): void {
         this.isExtension = isExtension;
     }
+
+    private get isPreloaderVisible(): boolean {
+        return (
+            this.container.querySelector("#preloader") !== null ||
+            this.container.querySelector("#preloader") !== undefined
+        );
+    }
+
+    private set isPreloaderVisible(visible: boolean) {
+        const preloader = this.container.querySelector("#preloader");
+        if (visible && !preloader) {
+            const preloaderElement = <HTMLElement>(
+                preloaderTemplate.content.cloneNode(true)
+            );
+            this.container.appendChild(preloaderElement);
+        } else if (!visible && preloader) {
+            preloader.remove();
```

Comment:
That's unfortunate but we need to use `preloader.parentNode!.removeChild(preloader);` here because some older libraries override `Element.remove()` (#3096). This usually occurs [on legacy pages](https://www.nytimes.com/elections/2008/results/president/map.html) (here, this may work the first time the page is loaded, but should crash after the page is refreshed).